nautobot_circuit_maintenance/handle_notifications/sources.py

```diff
@@ -193,9 +215,53 @@ def receive_notifications(
         self.close_session()
         return received_notifications
 
+    def validate_providers(self, logger: Job, notification_source: NotificationSource, since_txt: str) -> bool:
+        """Method to validate that the NotificartionSource has attached Providers.
+
+        Args:
+            logger (Job): Job to use its logger
+            notification_source (NotificationSource): Notification Source to validate providers
+            since_txt (str): Date string to be used to log
+
+        Returns:
+            bool: True if there are relevant providers attached or False otherwise
+        """
+        providers_with_email = []
+        providers_without_email = []
+        if not notification_source.providers.all():
+            logger.log_warning(
+                message="Skipping this email account no providers were defined.",
+            )
+            return False
+
+        for provider in notification_source.providers.all():
+            for custom_field, value in provider.get_custom_fields().items():
+                if custom_field.name == "emails_circuit_maintenances" and value:
+                    self._emails_to_fetch.extend(value.split(","))
+                    providers_with_email.append(provider.name)
+                else:
+                    providers_without_email.append(provider.name)
```

Comment:
This logic seems incorrect as it will add a provider that has multiple custom fields to both `providers_with_email` and `providers_without_email`.

```suggestion
            custom_fields = provider.get_custom_fields()
            emails = custom_fields.get("emails_circuit_maintenance")
            if emails:
                self._emails_to_fetch.extend(emails.split(","))
                providers_with_email.append(provider.name)
            else:
                providers_without_email.append(provider.name)
```

Reply:
yes, I agree with you